Website/src/rendering/src/components/Speakers/FeaturedSpeakers.tsx

```diff
@@ -1,50 +1,55 @@
-import Link from 'next/link';
-import { Text, Image } from '@sitecore-jss/sitecore-jss-nextjs';
+import { Text, Field, Image } from '@sitecore-jss/sitecore-jss-nextjs';
 import { ComponentProps } from 'lib/component-props';
 import { GraphQLSpeaker } from 'src/types/speaker';
+import Link from 'next/link';
 
 export type FeaturedSpeakersProps = ComponentProps & {
   fields: {
     data: {
+      source: {
+        numberOfSpeakers: Field<string>;
+      };
       item: {
         children: {
           results: GraphQLSpeaker[];
         };
       };
     };
   };
-  params: {
-    NumberOfSpeakers: string;
-  };
 };
 
 const FeaturedSpeakers = (props: FeaturedSpeakersProps): JSX.Element => {
-  const speakers =
-    props.fields.data?.item?.children?.results &&
-    props.fields.data.item.children.results
-      .filter((item) => item.featured.value)
-      .sort()
-      .slice(0, parseInt(props.params.NumberOfSpeakers))
-      .map((speaker, index) => (
-        <Link key={index} href={`/speakers/${speaker.itemName}`} passHref>
-          <a className="grid-item">
-            <Image
-              field={speaker.picture.jsonValue}
-              alt={speaker.name}
-              width={265}
-              height={265}
-              loading="lazy"
-            />
+  const speakers = props.fields.data.item.children.results
+    .filter((item) => item.featured.value)
+    .sort()
+    .slice(0, parseInt(props.fields.data.source?.numberOfSpeakers?.value))
```

Comment:
it has a default value so should never be broken...

Reply:
When a datasource item is created, I agree it had a default value of 6.
What I am talking about is what happens when you delete this default value of 6 from the field, save the datasource item, publish it, and load the page?
What also happens when it is set to 0 or a negative number?
We should write code to cover for those cases or add field validations inside Sitecore to prevent a user from setting an invalid value.